Revises exercises for the Interactive jobs tutorial page (#114)
AaltoSciComp/scicomp-docs: triton/tut/interactive.rst

Exercises
=========

The scripts you need for the following exercises can be found here:
`hpc-examples/slurm
<https://github.com/AaltoSciComp/hpc-examples/tree/master/slurm>`__)
You can clone the repository by running

> ``git clone https://github.com/AaltoSciComp/hpc-examples.git``

Reviewer: ```suggestion
``git clone https://github.com/AaltoSciComp/hpc-examples.git``.  This repository
will be used for the rest of the tutorial exercises.
```

1. The program ``hpc-examples/slurm/memory-hog.py``
   uses up a lot of memory to do nothing.  Let's play with it.
   It's run as follows:
   ``python hpc-examples/slurm/memory-hog.py 50M``, where the

>    last argument is however much memory you want to eat.

Reviewer: ```suggestion
   last argument is however much memory you want to eat.  You can use
   ``--help`` to see the options of the program.
```
I hope you can use --help - can we?

Author: Yes, we can.

   a) Try running the program with ``50M``.

   b) Run the program with ``50M`` and ``srun --mem=500M``.

>    c) Increase the amount of memory allocated.  How much memory can
>       you use before the job fails?

Reviewer: ```suggestion
   c) Increase the amount of memory the Python process tries to use (not the
      amount of memory Slurm allocates).  How much memory can
      you use before the job fails?
```

>    d) Look at the job history using ``slurm history`` - can you see
>       how much memory it actually used? (Try using the ``--sleep``
>       option)

Reviewer: ```suggestion
   d) Look at the job history using ``slurm history`` - can you see
      how much memory it actually used?
      
      Note, Slurm only measures memory every 60 seconds or so.  To make the program last longer, so that can be measured, give the ``--sleep`` option to the Python process, like this: ``python memory-hog.py 50M --sleep=60``.
```
is this the right syntax for --sleep?

Author: Yes, that's it.

2. The program ``hpc-examples/slurm/pi.py``
   calculates Pi using a simple stochastic algorithm.  The program takes
   one positional argument: the number of trials.

   The ``time`` program allows you to time any program,  e.g. you can
   ``time python x.py`` to print the amount of time it takes.

   a) Run the program, timing it with ``time``, a few times,
      increasing the number of trials, until it takes about 10
      seconds: ``time python hpc-examples/slurm/pi.py
      500`` and so on.

   b) Add ``srun`` in front (``srun python ...``).  Use the ``seff <jobID>``
      command to see how much time the program took to run.
      (If you'd like to use the ``time`` command, you can run

>       ``srun --mem=<m> --time=<t> bash -c
>       "time python time python hpc-examples/slurm/pi.py <iters>"``)

Reviewer: ```suggestion
      ``srun --mem=<m> --time=<t>time python time python hpc-examples/slurm/pi.py <iters>``)
```
I"m not sure about this, but raw `time` also works, it's just a different program!  I think it might be better to gloss over the differenet between `time` interactively and `time` here, than introduce something else that people might have to think about.  What do you tihnk?

Author: I was hesitant to add this but I thought it may be good to just let it be there so that anyone can try on their own. Because we started off with `time` here, I thought it's good to leave the option to use this too.

Reviewer: Since I'll be teaching this tomorrow, I'll add this.  I'd rather say "look, notice the time tool is different... someday you'll figure this out" than "don't worry about this weird syntax, it doesn't matter now but it's for (reason)".
